test/test_implicit_free_surface_solver.jl

```diff
@@ -25,9 +25,16 @@ function set_simple_divergent_velocity!(model)
     v .= 0
     η .= 0
 
-    imid = Int(floor(grid.Nx / 2)) + 1
-    jmid = Int(floor(grid.Ny / 2)) + 1
-    CUDA.@allowscalar u[imid, jmid, 1] = 1
+    # pick a point to set a non-zero u velocity
+    i = Int(floor(grid.Nx / 2)) + 1
+    j = Int(floor(grid.Ny / 2)) + 1
+    k = grid.Nz
+ 
+    grid isa ImmersedBoundaryGrid &&
+        grid.immersed_boundary.bottom_height[i, j] ≥ grid.underlying_grid.zᵃᵃᶜ[k] &&
+            @error "The point to set u is on land!"
```

Comment:
Would something like

```julia
inactive_cell(i, j, k, grid) && error("The nudged cell at ($i, $j, $k) is inactive.")
```

work?

Reply:
This catches both halos and immersed cells, and doesn't require so many explicit references to grid properties that might have to be changed / maintained in the future if we change the names of those properties.

Reply:
That's great suggestion. I did it!

Reply:
The problem persists though: the cell I'm nudging is **not** inactive but still I get η=0....